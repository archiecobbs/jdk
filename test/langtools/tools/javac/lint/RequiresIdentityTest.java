/*
 * @test /nodynamiccopyright/
 * @bug 8354556
 * @summary Expand value-based class warnings to java.lang.ref API
 * @compile/fail/ref=RequiresIdentityTest.out --patch-module java.base=${test.src} -Werror -XDrawDiagnostics -Xlint:identity RequiresIdentityHelper.java RequiresIdentityTest.java
 */

package java.lang;

public class RequiresIdentityTest extends RequiresIdentityHelper {
    class Box<T> {}

    RequiresIdentityHelper<Integer> field;                      // should warn
    RequiresIdentityHelper<Integer>[] field2;                   // should warn
    Box<? extends RequiresIdentityHelper<Integer>> field3;      // should warn
    Box<? super RequiresIdentityHelper<Integer>> field4;        // should warn

    public RequiresIdentityTest() {}
    public RequiresIdentityTest(Integer i) {
        super(i); // should warn
    }

    void test(RequiresIdentity2<Object> ri, Integer i) {
        RequiresIdentityHelper<Integer> localVar;     // should warn
        RequiresIdentityHelper<Integer>[] localVar2;  // should warn
        // there should be warnings for the invocations below
        ri.foo(i);
        ri.bar(i, i); // two for this one
        ri.gg(i);
    }
}
