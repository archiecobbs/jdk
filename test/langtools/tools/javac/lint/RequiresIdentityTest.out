RequiresIdentityTest.java:12:65: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:13:88: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:16:27: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:17:36: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:18:8: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:19:8: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:20:72: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:20:27: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:24:15: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:27:32: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:28:31: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:29:40: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:31:16: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:32:16: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:33:16: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:34:15: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:37:67: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:40:63: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:43:27: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:48:6: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:48:49: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:50:21: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:52:36: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:55:32: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:56:46: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:60:54: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:65:37: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:68:37: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:68:18: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:75:32: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:79:90: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:80:52: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:81:82: compiler.warn.attempt.to.use.value.based.where.identity.expected
RequiresIdentityTest.java:81:31: compiler.warn.attempt.to.use.value.based.where.identity.expected
- compiler.err.warnings.and.werror
- compiler.note.unchecked.filename: RequiresIdentityTest.java
- compiler.note.unchecked.recompile
1 error
34 warnings
